lightning/src/ln/peers/handler.rs

```diff
@@ -503,145 +557,118 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 			let pause_read = match peers.peers.get_mut(peer_descriptor) {
 				None => panic!("Descriptor for read_event is not already known to PeerManager"),
 				Some(peer) => {
-					assert!(peer.pending_read_buffer.len() > 0);
-					assert!(peer.pending_read_buffer.len() > peer.pending_read_buffer_pos);
 
-					let mut read_pos = 0;
-					while read_pos < data.len() {
-						{
-							let data_to_copy = cmp::min(peer.pending_read_buffer.len() - peer.pending_read_buffer_pos, data.len() - read_pos);
-							peer.pending_read_buffer[peer.pending_read_buffer_pos..peer.pending_read_buffer_pos + data_to_copy].copy_from_slice(&data[read_pos..read_pos + data_to_copy]);
-							read_pos += data_to_copy;
-							peer.pending_read_buffer_pos += data_to_copy;
+					let mut send_init_message = false;
+
+					let data_processing_decision = peer.encryptor.process_peer_data(data, &mut peer.pending_outbound_buffer);
+					match data_processing_decision {
+						PeerDataProcessingDecision::Disconnect(e) => {
+							log_trace!(self.logger, "Invalid act message; disconnecting: {}", e);
+							return Err(e);
 						}
 
-						if peer.pending_read_buffer_pos == peer.pending_read_buffer.len() {
```

Comment:
During the handshake `pending_read_buffer` is initialized to an act_length vector filled with 0s. So, until the full act arrives this branch will not be taken. This is a different behavior than this PR changes which return an `Err` if the `process_act()` doesn't receive the full act and results in a `PeerDataProcessingDecision::Disconnect`.

#672